src/main/java/org/example/service/BookService.java

```diff
@@ -0,0 +1,102 @@
+    }
+
+    public void updateBooks(List<Book> bookList) {
+        bookList.stream().filter(book -> book.isOrganizing())
+                .forEach(book -> book.setState(BookState.POSSIBLE));
+        this.bookRepository.updateList(bookList);
+    }
+    public Book createBook(String title, String author, int pageNum) {
+        Book book = new Book(bookRepository.getSize()+1, title, author, pageNum,BookState.POSSIBLE);
+
+        System.out.println("[System] 도서 등록이 완료되었습니다.\n");
+        bookRepository.addBook(book);
+
+        return book;
+    }
+
+    public void printAllBooks(List<Book> bookList) {
+        bookList.forEach(book -> {System.out.println(book.printBook());});
+    }
+    public List<Book> getAllBooks() {
+        return bookRepository.getAllBooks();
+    }
+
+    public List<Book> findByTitle(String word) {
+        return bookRepository.findByTitle(word);
+    }
+
+    public Book rentBook(int bookId) {
+        Book findBook = bookRepository.findById(bookId);
+        BookState findBookState = findBook.getState();
+
+        if(findBookState.equals(BookState.POSSIBLE)) {
+            findBookState.showChangeState();
+            findBook.setState(BookState.RENTING);
+        }else findBookState.showState();
+
+        return findBook;
+    }
+
+    public Book returnBook(int bookId) {
+        Book findBook = bookRepository.findById(bookId);
+        BookState findBookState = findBook.getState();
+
+        if (findBookState.equals(BookState.RENTING) || findBookState.equals(BookState.LOST)) {
+            BookState.RENTING.showChangeState();
+            findBook.setState(BookState.ORGANIZING);
+            // 도서 상태변화
+            Timer timer = new Timer(true);
+            timer.schedule(new UpdateTask(findBook), 10000);
+        } else {
+            findBookState.showState();
+        }
+
+        return findBook;
+    }
+
+    public Book lostBook(int bookId) {
+        Book findBook = bookRepository.findById(bookId);
+        BookState findBookState = findBook.getState();
+
+        if (!findBookState.equals(BookState.LOST)) {
+            BookState.LOST.showChangeState();
+            findBook.setState(BookState.LOST);
+        } else {
+            findBookState.showState();
+        }
+
+        return findBook;
+    }
+
+    public Book deleteBook(int bookId) {
+
+        if(bookId > bookRepository.getSize()) {
+            System.out.println("[System] 존재하지 않는 도서번호 입니다.");
+            throw new NoSuchElementException("존재하지 않는 도서번호 입니다.");
+        }else {
+            bookRepository.updateListId(bookId);
+            Book deleteBook = bookRepository.deleteById(bookId);
+            System.out.println("[System] 도서가 삭제 처리 되었습니다.\n");
+            return deleteBook;
+        }
+    }
+
+    public String getByText(String text) {
+        BookState bookState = Arrays.stream(BookState.values()).filter(state -> state.toString().equals(text))
+                .findFirst().orElseThrow(() -> new RuntimeException());
```

Comment:
```suggestion
        BookState bookState = Arrays.stream(BookState.values()).filter(state -> state.toString().equals(text))
                .findFirst()
                .orElseThrow(() -> new RuntimeException());
```

1. 한줄에 점 하나씩 쓰면 가독성이 높습니다.
2. RunTimeExcpetion은 에러중에도 꽤 상위에 있어서 에러 클래스만 보면 나중에 로그에서 대체 어떤 에러때문이야? 코드를 다 뒤져야 하거든요... 표준에러중에 맞는 에러를 써보시고, 메시지도 넣어보시면 어떨까요?